### DISCUSSION TITLE
Coupling multiple domains into a source term

### DISCUSSION THREAD

**Discussion by AnthonyB08** [idaholab/moose#18709] [Q&A General] [answered]
what source files would I utilize to impose a source term on an interface boundary that is coupled with the variable in the neighboring domain? See figure for governing problem

![image](https://user-images.githubusercontent.com/55299169/131040805-c4fac34b-214d-454d-95dc-d0427567da54.png)

So far, I am thinking I will need to use Auxvariable to hold the value of the P interface and pass it into the source kernel of the domain.

**Comment by GiudGiud** [comment, marked as answer]
Hello 

You should have a look at InterfaceKernels. 
What kind of source is it? Diffusive? A current proportional to a the value of the variable?

Guillaume

**Comment by AnthonyB08** [comment]
it is a permeation source term. Where the accumulation or depletion of the species is dependent on the interface variables i.e pressure .

**Reply by GiudGiud** [reply]
ok seems like you will want to work with an interface kernel then. Note that you can have some of the coefficients be defined completely in the kernel or be defined as material properties, depending on what is most convenient for you

**Reply by AnthonyB08** [reply]
Could you elaborate more on the type of interface kernels, I.e what specific class files would you recommend. I am still in the process of learning moose.

**Reply by AnthonyB08** [reply]
Also I do not believe interface kernels are the route, because I would need to describe the source term (permeation) in the respected domain with its dependency of value on the respected interface ( pressure).

**Reply by GiudGiud** [reply]
the permeation term, is it a surface term or a volumetric one?

**Reply by AnthonyB08** [reply]
Surface Term ( mass transfer of u1 into domain 2) k*(c-c^*)

**Reply by GiudGiud** [reply]
I would look at the conjugate heat transfer interface kernel.
If it werent for that pressure dependence (variables from the other side of the interface right?) I would say a boundary condition would work.
